engine/baml-lib/schema-ast/src/parser/datamodel.pest

```diff
@@ -34,7 +38,7 @@ value_expression          = { identifier ~ expression? ~ (NEWLINE? ~ field_attri
 type_builder_block = {
     TYPE_BUILDER_KEYWORD ~ BLOCK_OPEN ~ type_builder_contents ~ BLOCK_CLOSE
 }
-type_builder_contents = { (type_expression_block | type_alias | comment_block | empty_lines | BLOCK_LEVEL_CATCH_ALL)* }
+type_builder_contents = { (dynamic_type_expression_block | type_expression_block | type_alias | comment_block | empty_lines | BLOCK_LEVEL_CATCH_ALL)* }
```

Comment:
We still want to support `type_expression_block` in `type_builder_contents`? (As opposed to replacing `type_expression_block` with `dynamic_type_expression_block`?)